src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPoolManager.cs

```diff
@@ -527,6 +556,38 @@ private static string GetIdentityIfDefaultCredentialsUsed(bool defaultCredential
             return defaultCredentialsUsed ? CurrentUserIdentityProvider.GetIdentity() : string.Empty;
         }
 
+        public static int GetMaxHttp11ConnectionsPerPool()
+        {
+            int max = 0;
+            foreach ((WeakReference<HttpConnectionPoolManager> weakConnectionPoolManagerRef, _) in s_allConnectionPools)
+            {
+                if (weakConnectionPoolManagerRef.TryGetTarget(out HttpConnectionPoolManager? connectionPoolManager))
+                {
+                    foreach ((_, HttpConnectionPool connectionPool) in connectionPoolManager._pools)
+                    {
+                        max = Math.Max(max, connectionPool.Http11ConnectionCount);
+                    }
+                }
+            }
+            return max;
+        }
```

Comment:
Instead of maintaining a collection and looping, can you modify `HttpConnectionPool` to call a `HttpTelemetry.SetNewMaximum(x)` whenever it opens up a new connection, and `Http2Connection` to do similar when it opens a new stream? It should be much cheaper this way.

Reply:
I still question the value of these max counters. Wouldn't a current count be more useful? You could approximate a max on top of it, but you can't do the inverse.

Reply:
I agree. A current count seems a lot more useful.

Reply:
> can you modify HttpConnectionPool to call a HttpTelemetry.SetNewMaximum(x) whenever it opens up a new connection, and Http2Connection to do similar when it opens a new stream?

The maximum reported here can fluctuate up and down, from what I understand that approach would only allow growing the max over time.

I also don't see the value of this information. @davidsh if you have the time, could you share what use case you had in mind?

By "current count", I assume you mean a sum of all opened http11 connections / all active Http2 streams? (in which case this global list mess isn't needed since we can just increment/decrement a global counter)

Reply:
> By "current count", I assume you mean a sum of all opened http11 connections / all active Http2 streams?

Yup

Reply:
> By "current count", I assume you mean a sum of all opened http11 connections / **all active Http2 streams**

Does it really make sense to count total number of streams on all connections? I think it would be better to count HTTP/2 connections only. HTTP/2 stream is a logical entity in the nutshell whereas HTTP/2 connection is a physical one consuming limited resources (in example SNAT ports).